tools/azsdk-cli/Azure.Sdk.Tools.Cli/Services/Languages/GoLanguageRepoService.cs

```diff
@@ -0,0 +1,135 @@
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Text;
+using Microsoft.Extensions.Logging;
+using Azure.Sdk.Tools.Cli.Models;
+using Azure.Sdk.Tools.Cli.Helpers;
+
+namespace Azure.Sdk.Tools.Cli.Services;
+
+/// <summary>
+/// Go-specific implementation of language repository service.
+/// Uses tools like go build, go test, go mod, gofmt, etc. for Go development workflows.
+/// </summary>
+public class GoLanguageRepoService : LanguageRepoService
+{
+    private readonly ILogger<GoLanguageRepoService> _logger;
+    private readonly string compilerName = "go";
+    private readonly string formatterName = "goimports";
+    private readonly string linterName = "golangci-lint";
+
+    public GoLanguageRepoService(IProcessHelper processHelper, IGitHelper gitHelper, ILogger<GoLanguageRepoService> logger) 
+        : base(processHelper, gitHelper)
+    {
+        _logger = logger;
+        
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+        {
+            compilerName = "go.exe";
+            formatterName = "gofmt.exe";
+            linterName = "golangci-lint.exe";
+        }
+    }
+
+    #region Go specific functions, not part of the LanguageRepoService
+
+    public async Task<CLICheckResponse> CreateEmptyPackage(string packagePath, string moduleName)
```

Comment:
nit: You can have an abstra method in LanguageRepoService that can return not implemented exception and override it only in GoLanguageService

Reply:
Let's tag that one for the next PR.